src/languages/fsharp.js

```diff
@@ -94,30 +99,21 @@ export default function(hljs) {
     illegal: /\/\*/,
     contains: [
       {
-        // monad builder keywords (matches before non-bang kws)
-        className: 'keyword',
-        begin: /\b(yield|return|let|do)!/
-      },
-      {
-        className: 'string',
-        begin: '@"',
-        end: '"',
-        contains: [
-          {
-            begin: '""'
-          }
-        ]
+        // monad builder keywords (matches before non-bang keywords)
+        scope: 'keyword',
+        match: /\b(yield|return|let|do|match|use)!/
       },
       {
-        className: 'string',
-        begin: '"""',
-        end: '"""'
+        scope: 'string',
+        // matches triple quote strings, verbatim strings (@""), character literals...
+        match: /(?:"""[\s\S]*?"""|@"(?:""|[^"])*"|"(?:\\[\s\S]|[^\\"])*")|'(?:[^\\']|\\(?:.|\d{3}|x[a-fA-F\d]{2}|u[a-fA-F\d]{4}|U[a-fA-F\d]{8}))'/
       },
-      hljs.COMMENT('\\(\\*(\\s)', '\\*\\)', {
+      hljs.COMMENT(/\(\*(?!\))/, /\*\)/, {
         contains: ["self"]
       }),
       {
-        className: 'class',
+        // type definitions:
+        scope: 'title.class',
```

Comment:
`title.class` is for names of classes such as `WordParser`, etc... (and `class` alone is deprecated)  you may want a multi-matcher here... what is the syntax you're trying to cover?

Reply:
I changed this because I noticed `class` was deprecated. I thought `title.class` was the replacement.

The syntax I'm trying to cover is any type definition:
```fsharp
type A = int * int // type abbreviation
type B = { FirstName: string; LastName: string } // record
type C = Circle of int | Rectangle of int * int // discriminated union
type D = Day | Month | Year // another discriminated union
type E<'a> = Choice1 of 'a | Choice2 of 'a * 'a // a generic type definition

type MyClass(initX: int) = // class
   let x = initX
   member _.Method() = printf "x=%i" x
```
It's currently parsed as follows:
![image](https://user-images.githubusercontent.com/1535823/135766230-2cceda37-b58d-4192-ac42-a464f4570e7e.png)

I'm not sure I understand the meaning behind all the different standard scopes. What should I do?

Reply:
> The syntax I'm trying to cover is any type definition (the part before the first = or ():

We do not  care/desire this type of "semantic" flagging (that what we deprecated when we deprecated `class`).  IE we don't care about "class definitions".  We only care about "class name/title" (which may or may not be in a definition).   So above there should be NO nesting. 

- `type` is a keyword 
- `A` is a `title.class`
- `type A = int * int` (overall is nothing)